app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/widget/MVVWidgetConfigureActivity.kt

```diff
@@ -0,0 +1,194 @@
+        val intent = intent
+        val extras = intent.extras
+        if (extras != null) {
+            appWidgetId = extras.getInt(
+                    AppWidgetManager.EXTRA_APPWIDGET_ID,
+                    AppWidgetManager.INVALID_APPWIDGET_ID)
+        }
+
+        val tm = TransportController(this)
+        this.widgetDepartures = tm.getWidget(appWidgetId)
+
+        val autoReloadSwitch = findViewById<Switch>(R.id.mvv_widget_auto_reload)
+        autoReloadSwitch.isChecked = this.widgetDepartures.autoReload
+        autoReloadSwitch.setOnCheckedChangeListener { _, checked ->
+            widgetDepartures.autoReload = checked
+        }
+        // TODO add handling for use location
+
+        listViewResults = findViewById(R.id.activity_transport_listview_result)
+        listViewResults.onItemClickListener = this
+
+        // Initialize stations adapter
+        val recentStations = recentsDao.getAll(RecentsDao.STATIONS) ?: emptyList()
+        adapterStations = ArrayAdapter(this, android.R.layout.simple_list_item_1,
+                TransportController.getRecentStations(recentStations))
+
+        if (adapterStations.count == 0) {
+            openSearch()
+        } else {
+            listViewResults.adapter = adapterStations
+            listViewResults.requestFocus()
+        }
+    }
+
+    /**
+     * Click on station in list
+     */
+    override fun onItemClick(av: AdapterView<*>, v: View, position: Int, id: Long) {
+        val (station, id1) = av.adapter.getItem(position) as StationResult
+        widgetDepartures.station = station
+        widgetDepartures.stationId = id1
+        saveAndReturn()
+    }
+
+    override fun onStartSearch() {
+        val recents = recentsDao.getAll(RecentsDao.STATIONS)
+        if (recents == null) {
+            listViewResults.adapter = NoResultsAdapter(this)
+            listViewResults.requestFocus()
+            return
+        }
+
+        val stations = TransportController.getRecentStations(recents)
+        displayStations(stations)
+    }
+
+    override fun onStartSearch(query: String) {
+        disposable.add(TransportController.getStationsFromExternal(this, query)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(this::displayStations) {
+                    // Something went wrong
+                    Utils.log(it)
+                    onStartSearch()
+                })
+    }
+
+    private fun displayStations(stations: List<StationResult>) {
+        showLoadingEnded()
+
+        if (stations.isEmpty()) {
+            // So show no results found
+            listViewResults.adapter = NoResultsAdapter(this)
+            listViewResults.requestFocus()
+            return
+        }
+
+        adapterStations.clear()
+        adapterStations.addAll(stations)
+
+        adapterStations.notifyDataSetChanged()
+        listViewResults.adapter = adapterStations
```

Comment:
I think this line is unnecessary. Via the three lines above, the items in the list view should already be updated. Or am I missing something?